File: templates/cs/Utility.cs
/// <summary>
/// Utility class with static methods
/// </summary>
static class Utility
{
  public const string Separator = "---------------";

  /// <summary>
  /// Writes a message to the console and returns the user's input
  /// </summary>
  /// <param name="message"></param>
  /// <typeparam name="Type"></typeparam>
  /// <returns></returns>
  public static Type Prompt<Type>(string message)
  {
    Console.WriteLine(message);

    Type result = (Type)Convert.ChangeType(Console.ReadLine(), typeof(Type))!;

    if (result == null)
    {
      throw new FormatException("Invalid input");
    }

    return result;
  }

  /// <summary>
  /// Prints a header in the console using the separator constant
  /// </summary>
  /// <param name="message"></param>
  public static void PrintHeader(string message)
  {
    Console.WriteLine(Separator);
    Console.WriteLine(message);
    Console.WriteLine(Separator);
  }
}

File: templates/java/Utility.java
import java.util.Scanner;

public class Utility {
    public static final String SEPARATOR = "----------------------------------------";

    /**
     * Prompts the user for input with a message.
     *
     * @param   message The message to display to the user.
     * @param   type    The type of input to expect.
     * @return          User input.
     */
    public static <T> T Prompt(String message, Class type) throws Exception {
        Scanner scanner = new Scanner(System.in);

        System.out.println(message);

        if (type == String.class) {
            return (T) scanner.nextLine();
        } else if (type == Integer.class) {
            return (T) Integer.valueOf(scanner.nextLine());
        } else if (type == Float.class) {
            return (T) Float.valueOf(scanner.nextLine());
        } else if (type == Double.class) {
            return (T) Double.valueOf(scanner.nextLine());
        } else {
            throw new Exception("Invalid type");
        }
    }

    /**
     * Prints a message to the console and wraps it in a separator.
     *
     * @param   message message to be printed
     */
    public static void PrintHeader(String message) {
        System.out.println(Utility.SEPARATOR);
        System.out.println(message);
        System.out.println(Utility.SEPARATOR);
    }
}
